spec.html

```diff
@@ -32457,6 +32591,120 @@ <h1>Expanded Years</h1>
+          </dl>
+          <emu-alg>
+            1. Let _parseResult_ be ParseText(StringToCodePoints(_offsetString_), |UTCOffset|).
+            1. If _parseResult_ is a List of errors, return *false*.
+            1. Return *true*.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-parsetimezoneoffsetstring" type="abstract operation">
+          <h1>
+            ParseTimeZoneOffsetString (
+              _offsetString_: a String,
+            ): an integer
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>The return value is the UTC offset, as a number of nanoseconds, that corresponds to the String _offsetString_.</dd>
+          </dl>
+          <emu-alg>
+            1. Let _parseResult_ be ParseText(StringToCodePoints(_offsetString_), |UTCOffset|).
+            1. Assert: _parseResult_ is not a List of errors.
+            1. Assert: _parseResult_ contains a |TemporalSign| Parse Node.
+            1. Assert: _parseResult_ contains an |Hour| Parse Node.
+            1. Let _parsedSign_ be the source text matched by the |TemporalSign| Parse Node contained within _parseResult_.
+            1. If _parsedSign_ is the single code point U+002D (HYPHEN-MINUS) or U+2212 (MINUS SIGN), then
+              1. Let _sign_ be -1.
+            1. Else,
+              1. Let _sign_ be 1.
+            1. Let _parsedHours_ be the source text matched by the |Hour| Parse Node contained within _parseResult_.
+            1. Let _hours_ be the MV of CodePointsToString(_parsedHours_).
+            1. If _parseResult_ does not contain a |MinuteSecond| Parse Node, then
+              1. Let _minutes_ be 0.
+              1. Let _seconds_ be 0.
+            1. Else,
+              1. Let _parsedMinutes_ be the source text matched by the first |MinuteSecond| Parse Node contained within _parseResult_.
+              1. Let _minutes_ be the MV of CodePointsToString(_parsedMinutes_).
+            1. If _parseResult_ does not contain two |MinuteSecond| Parse Nodes, then
+              1. Let _seconds_ be 0.
+            1. Else,
+              1. Let _parsedSeconds_ be the source text matched by the second |MinuteSecond| Parse Node contained within _parseResult_.
+              1. Let _seconds_ be the MV of CodePointsToString(_parsedSeconds_).
+            1. If _parseResult_ does not contain a |TemporalDecimalFraction| Parse Node, then
+              1. Let _nanoseconds_ be 0.
+            1. Else,
+              1. Let _parsedFraction_ be the source text matched by the |TemporalDecimalFraction| Parse Node contained within _parseResult_.
+              1. Let _fraction_ be the string-concatenation of CodePointsToString(_parsedFraction_) and the string *"000000000"*.
+              1. Let _nanosecondsString_ be the String value equal to the substring of _fraction_ from 1 to 10.
+              1. Let _nanoseconds_ be the MV of _nanosecondsString_.
```

Comment:
I guess we could just use the parse node that we already get from the parse tree returned from ParseText(..., |UTCOffset|)?

Reply:
(Except in this nanoseconds case)

Reply:
> I guess we could just use the parse node that we already get from the parse tree returned from ParseText(..., |UTCOffset|)?

The types line up that way, but to make it work in full you'd also have to define MV over all of those productions - right now [its definition](https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#sec-static-semantics-mv) does not know how to handle an `|Hour|` or `|MinuteSecond|` parse node. And adding those would be kind of annoying.

Now that I'm looking at this again, I think the best thing would be to use [StringToNumber](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-stringtonumber) in all cases (including the nanoseconds), and then cast to a real. This only works because you know these are small integers (i.e. less than MAX_SAFE_INTEGER), but I think that fact is obvious enough in these contexts. (Sorry I didn't suggest this originally; I hadn't noticed the other uses of MV.)

It's possible the other editors will prefer the re-parsing approach, though. I'll ask tomorrow and get this resolved. And don't worry about fixing it up; once we decide what approach to use the editors can make the last tweaks and land it.

Reply:
I also prefer using StringToNumber here with a note that it's not lossy in this case. DateTime stuff tend to get implemented line-for-line from the spec text, so using StringToNumber here would aid implementation clarity.

Reply:
Pushed up 1688afc59b5aa5c6757e96fc13f5f4d16ad6a3d3 switching to StringToNumber.